milli/src/search/query_tree.rs

```diff
@@ -75,9 +76,13 @@ impl Operation {
         }
     }
 
-    fn phrase(mut words: Vec<String>) -> Self {
+    fn phrase(mut words: Vec<Option<String>>) -> Self {
         if words.len() == 1 {
-            Self::Query(Query { prefix: false, kind: QueryKind::exact(words.pop().unwrap()) })
+            if let Some(word) = words.pop().unwrap() {
+                Self::Query(Query { prefix: false, kind: QueryKind::exact(word) })
+            } else {
+                Self::Phrase(words)
+            }
```

Comment:
It was crashing because of `phrase.windows` here: https://github.com/meilisearch/milli/blob/b34fbcf4ebb2144e538f774dcb67d95581d66f6e/milli/src/search/criteria/mod.rs#L350 (that function panics when the length of the slice is 0).

Fixed this in 655075df3fa943aef645e7b2cb8c5860eb4d17e2 and added a test for it.